assemblyscript/assembly/as-wasi-nn.ts

```diff
@@ -127,16 +164,36 @@ export class Tensor {
     }
 }
 
+/**
+ * The type of data contained within a `Tensor`.
+ */
 export const enum TensorType {
     f16 = 0,
     f32 = 1,
     u8 = 2,
     i32 = 3
 }
 
+/**
+ * A wasi-nn failure.
+ */
 export class WasiNnError extends Error {
     constructor(message: string = "", code: i32 = -1) {
         super(message + "; error code = " + code);
         this.name = "WasiNnError";
     }
 }
+
+/**
+ * Helper function to capture the pointer to the beginning of an array.
+ * @param data an array
+ * @returns a pointer to the array data
+ */
+// @ts-ignore: decorator
+@inline
+function getArrayPtr<T>(data: T[]): u32 {
```

Comment:
for pointers better use `usize` which becomes `u32` in wasm32 and `u64` in wasm64 (in future)

Reply:
I tried that out and what ends up happening is that we have to cast everything back to `i32` to match the wasi-nn API. I sort of feel that, for now, it is clearer to leave everything in terms of `u32` and `i32` and fix both the API and these bindings once wasm64 is closer to being readily available.

Reply:
I agree with @abrown , lets keep it as u32 for now to prevent the need for a bunch of casting. We can easily change the u32 to usize once wasm64 lands.